src/scancode/resource.py

```diff
@@ -1392,7 +1415,8 @@ def __init__(self, location,
                  codebase_attributes=None,
                  full_root=False, strip_root=False,
                  temp_dir=temp_dir,
-                 max_in_memory=10000):
+                 max_in_memory=10000,
+                 *args, **kwargs):
```

Comment:
Just to be clear here: is your point that a virtualcodebase (e.g. --from-json) would never be subject to any max-depth filtering since it does not handle it at all? It may be OK, but it would need to be documented in the code and the CLI IMHO.

Reply:
Yeah, that is what I thought. In case of VirtualCodebase, the directories are already scanned with a particular depth so using --from-json would go to the same original depth. Or would it make more sense to specify a different depth while scanning from JSON?

Reply:
Or maybe we can notify when --max-depth is used with --from-json. A warning message or something.

Reply:
What you have now works for me.